src/clients/HttpTunneler.cc

```diff
@@ -374,11 +375,22 @@ Http::Tunneler::sendSuccess()
 }
 
 void
-Http::Tunneler::countFailingConnection()
+Http::Tunneler::countFailingConnection(ErrorState *error)
 {
     assert(connection);
-    if (const auto p = connection->getPeer())
-        peerConnectFailed(p);
+    if (const auto p = connection->getPeer()) {
+        std::unique_ptr<ACLFilledChecklist> ch;
+        if (Config.accessList.cachePeerFault) {
+            ch.reset(new ACLFilledChecklist(Config.accessList.cachePeerFault, request.getRaw(), nullptr));
+            ch->al = al;
+            ch->syncAle(request.getRaw(), nullptr);
+            if (error) {
+                ch->reply = error->response_.getRaw();
+                HTTPMSGLOCK(ch->reply);
+            }
```

Comment:
Here and in most other callers, should not we also set ACLFilledChecklist::dst_addr? We do know where the connection was going, right? Or does that setting already happen automatically elsewhere??

If that code is missing, it may be a good idea to add a method like ACLFilledChecklist::setToPeerConnection() or ACLFilledChecklist::setOutgoingConnection() that will extract that address (and possibly other details like dst_peer_name) from the supplied Comm::Connection object. If you add such a method, please make it safe for nil and half-baked/never-opened Connection objects -- the caller should not have to check what state that Connection object is.

N.B. ACLFilledChecklist::src_addr is set in ACLFilledChecklist::setRequest().